src/Lean/Elab/Tactic/Simp.lean

```diff
@@ -236,12 +244,52 @@ def mkSimpContext (stx : Syntax) (eraseLocal : Bool) (kind := SimpKind.simp) (ig
       unless simpTheorems.isErased localDecl.userName do
         let fvarId := localDecl.fvarId
         let proof  := localDecl.toExpr
-        let id     ← mkFreshUserName `h
-        fvarIdToLemmaId := fvarIdToLemmaId.insert fvarId id
-        simpTheorems ← simpTheorems.addTheorem proof (name? := id)
+        fvarIdToLemmaId := fvarIdToLemmaId.insert fvarId fvarId.name
```

Comment:
Why is this change necessary? You are using the internal free variable identifier as the theorem name. This name may appear in the trace. I prefer to have a `h???` than the unique free variable identifier.

Reply:
I found the explanation for this change, but we need to document it.